source/agora/node/Registry.d

```diff
@@ -512,14 +512,26 @@ private PublicKey extractPublicKey (in char[] domain) @safe
         catch (Exception exc) return PublicKey.init;
     }
 
-    if (auto str = domain.splitter('.').front)
+    auto splitted = domain.value.splitter('.');
+    if (auto str = splitted.front)
     {
-        if (str.length == PublicKeyStringLength)
-            return tryParse(str);
-        else if (str.length == NoHRPPublicKeyStringLength)
-            return tryParse("boa1" ~ str);
-        else
-            return PublicKey.init;
+        // Includes service and protocol labels
+        if (domain.serviceAndProtocolLabels() != Domain.ServiceProtoTuple.init)
+        {
+            splitted.popFront(); // Pop service
+            splitted.popFront(); // Pop proto
+            str = splitted.front;
+        }
+
+        if (str)
```

Comment:
I dont think this is needed, if `str == null`, `str.length` is 0

Reply:
(not so) fun fact:
```D
void main ()
{
 	string str = "";
    if (str)
        assert(0); // Triggers
}
```

That's why we should **always** use `.length` when checking for empty string, not `if (str)`, `if (str is null)`, or `if (str == string.init)`.

Reply:
Below lines are checking for `.length` already; thus removing